programs/tests/c-tests/gpio/gpio.c

```diff
@@ -48,5 +97,10 @@ int main() {
 
     printf("Set invalid bit has no effect\n");
 
+#if HAVE_EMULATOR_CLIENT
+    run_emulator_client_tests();
+    printf("Pins were toggled in a random-ish fashion test passed\n");
+#endif // HAVE_EMULATOR_CLIENT
```

Comment:
I think we should have an #error if we dont have the emulator client?

Reply:
The reason I made it like that was to avoid having to deal with tests failing in the CICD, since it currently does not support using clients for tests. I guess this is something it should have, but I did not take the time to implement it and kept it like that instead.

But I feel like adding clients for testing with the CICD is a nice to have and could maybe be done later?